Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java

```diff
@@ -81,17 +90,33 @@ private NetworkParameters parseNetworkFrom(Configuration.BitcoinConfig bitcoinCo
         }
     }
 
+    InputStream checkpoints;
+
     public void setupWalletFromMnemonic(List<String> mnemonicCode, @Nullable KeyParameter key) {
         DeterministicSeed seed = new DeterministicSeed(mnemonicCode, null, "", 1539388800);
         Runnable setupWalletFromBackup = () -> {
             try {
                 KeyChainGroup keyChainGroup = new KeyChainGroup(params, seed);
                 wallet = new Wallet(params, keyChainGroup); //Wallet.fromSeed(params, seed);
-                synchronizeWalletBlocking(wallet);
-                if (key != null)
-                    saveEncryptedWallet(key);
+                if (checkpoints == null)
+                    checkpoints = CheckpointManager.openStream(params);
+
+                if (checkpoints != null) {
+                    // Initialize the chain file with a checkpoint to speed up first-run sync.
+                    long time = seed.getCreationTimeSeconds();
+                    removeOldBlockStore();
+                    if (time > 0)
+                        CheckpointManager.checkpoint(params, checkpoints, blockStore, wallet.getEarliestKeyCreationTime());
+                    else
+                        Logger.info("Creating a new uncheckpointed block store due to a wallet with a creation time of zero: this will result in a very slow chain sync");
+
+
+                } else removeOldBlockStore();
+
+                synchronizeWalletBlocking(wallet, key);
+
                 Logger.info("Wallet balance" + wallet.getBalance().toFriendlyString());
-            } catch (IOException | BlockStoreException | WalletNotInitialized walletNotInitialized) {
+            } catch (IOException | BlockStoreException walletNotInitialized) {
                 walletNotInitialized.printStackTrace();
```

Comment:
`WalletNotInitialized` indicates that the operation that you want to perform is not allowed, until you initialize the wallet first. And the `setupWalletFromMnemonic` is the method responsible of the wallet initialization, so it musn't throw WalletNotInitialized because at this point should not be initialized, it was a bug.

Reply:
But yes `.printStackTrace()` is not the best way to handle, any exception. But I would fix it after main release.

Reply:
okay no problem